### PR TITLE
Use Instantiate for LTL intrinsic instantiation

### PR DESCRIPTION
**Pull request by jackkoenig**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Performance improvement

#### Desired Merge Strategy

- Squash

#### Release Notes

This vastly reduces the IR size when using LTL properties

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x`, `3.6.x`, or `5.x` depending on impact, API modification or big change: `6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by sequencer** [comment]
Any plan to give LTL IR and emit them in Chisel?

**Comment by jackkoenig** [comment]
There is a bug in ScalaDoc that is making this not work: https://stackoverflow.com/questions/42684101/scaladoc-generation-fails-when-referencing-methods-generated-by-annotation-macro

There are ways to mitigate that I am trying out but they are also not working for reasons I am unsure of. The debugging continues...

File: src/main/scala/chisel3/util/circt/LTLIntrinsics.scala [hunk 1]
```diff
@@ -100,3 +100,37 @@ private[chisel3] class VerifAssumeIntrinsic(label: Option[String]) extends Verif
...
 /** A wrapper intrinsic for the CIRCT `verif.cover` operation. */
 private[chisel3] class VerifCoverIntrinsic(label: Option[String]) extends VerifAssertLikeIntrinsic("cover", label)
+
+// There appears to be a bug in ScalaDoc where you cannot use macro-generated methods in the same compilation unit
+// as the macro-generated type. This means that any use of Definition[_] or Instance[_] of the above classes within
+// this compilation unit breaks ScalaDoc generation
+// This issue appears to be similar to https://stackoverflow.com/questions/42684101 but applying the specific mitigation
+// did not seem to work.
+// As a workaround, we simply write the extension methods that are generated by the @instantiable macro so that we can use
+// them here.
```
Comment (seldridge) [inline src/main/scala/chisel3/util/circt/LTLIntrinsics.scala:110]: Nit: weird text flow here.

**Review by seldridge — APPROVED** [review]

File: src/main/scala/chisel3/util/circt/LTLIntrinsics.scala (hunk 1 above)
Comment (jackkoenig) [reply src/main/scala/chisel3/util/circt/LTLIntrinsics.scala]: Fixed.

**Pull request merged**